### DISCUSSION TITLE
move variables / constants to a separate file for the accuracy of tests.

### DISCUSSION THREAD

**Discussion by Marat-Ka** [official-stockfish/Stockfish#3267] [General]
My question is: what do we have now: two assemblies from the same source in a row will have a different hash, i.e. they are not binary equal. This means that there is a very high probability that one of them will be 0.1 - 1% faster than the other. Again, it is quite possible to measure it, usually the difference is small, but sometimes it can be around that same 1%.

What follows from this:

If we run the tournament master vs patch, then if the "patch" is a little faster, simply due to the randomness of the compiler, then the patch can gain a certain plus, having in fact a minus. And this code will pass!

But if you put the variables / constants into a file, then the patches that change them (but not the code itself) can be run on the same binar, getting greater accuracy.

**Comment by Vizvezdenec** [comment]
We partially have it (a lot of constants are in types.h).
And I honestly don't find it all that amusing because as a developer you need often to look into a separate file to see what constant value there is, it's not always that great...
